content/posts/a-guide-on-commit-messages.mdx

```diff
@@ -0,0 +1,169 @@
+
+Commit messages are not only for your coworkers but also for your future self as Peter Hutterer said in this great article [On commit messages](http://who-t.blogspot.com/2009/12/on-commit-messages.html).
+
+> Any software project is a collaborative project. It has at least two developers, the original developer and the original developer a few weeks or months later when the train of thought has long left the station.
+>
+> Peter Hutterer - [On commit messages](http://who-t.blogspot.com/2009/12/on-commit-messages.html)
+
+A project is made to be maintained in the long term. Sometimes years after a piece of code has been written you will ask yourself why it has been written like that. It happened to me this week on function I coded only 6 months ago. With a well written commit message it is easier to get back in the state of mind at the time you wrote this change.
+
+Taking care of your commit messages can also simplify your daily life. Some tools becomes more interesting with a meaningful commit history like a few git commands `log`, `rebase`, `cherry-pick`, ... When you come back from holidays a simple `git log --pretty` will give you a lot of information about what your coworkers did while you was away. Writing good commit messages force you to split your change making it easier to review with `git log -p`.
+
+Speaking of review did you ever struggle rebasing your code on some change a developer made? It becomes easier to know why a change was made if the commit message carries real meaning. It makes `rebase` result much more predictible. Another example is `git blame` command which is by the way not made to make fun of your colleagues but giving you context on a line of code you come accross. It can be pretty useful when you want to refactor a piece of code. 
+
+Last but not least, a well structured commit history enables generating a changelog between two version of your project to inform your users what changed.
+
+This is a huge amount of reasons to start taking time to write good commit messages.
+
+## How to write a good commit message
+
+Now that you are convinced commit messages are important. Let's talk about how to write good ones. I picked commit messages from a project I work on called [immutadot](https://github.com/zenika-open-source/immutadot). As you can see these commit messages are not consistent. It makes them hard to read.
+
+```
+fix #72 unnecessary object copies
+Fixing unshift documentation examples
+seq.chain path parameter fix #18
+Remove built files
+Add tests
+Add Circle CI base config
+Rename es -> src
+Add lang package with toggle, fix exports
+Core package
+```
+
+In these commit message we can see a few mistakes in term of style, content and metadata. Chris beams describe well these three points in his article [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
```

Comment:
1. You should put an `s` to `message` => In these commit messages
2. You should put an `s` to `describe ` => `Chris Beams describes`